### PR TITLE
Bugfix/561 - facets limit not works

### PR DESCRIPTION
**Pull request by SamSalvatico**
Related to issue [561](https://github.com/oramasearch/orama/issues/561)

I fixed the casting of the facets config in code, that was preventing to get the correct property value from the set config.

Also added tests for both default and non default `limit`.

**Please Note**: I didn't find mention, in code, specifically in the declaration of `StringFacetDefinition`, to the `size` field, then I updated the docs removing it and replacing it with `limit`, where needed.

### CODE REVIEW

File: packages/orama/tests/facets.test.ts [hunk 1]
```diff
@@ -216,5 +216,75 @@ t.test('facets', (t) => {
...
+    const db = await create({
+      schema: {
+        author: 'string',
+        quote: 'string'
+      }
+    })
+
+    const quotes = [];
+
+    for (let i = 0; i <= orderedAuthors.length; i++) {
```
Comment (allevo) [inline packages/orama/tests/facets.test.ts:270]:
```suggestion
    for (let i = 0; i < orderedAuthors.length; i++) {
```
off by one

**Comment by allevo** [comment]
Thanks for your contribution. I don't understand the last test: the number of quotes is 11. Which expected behavior should we have when a limit of 20 is specified? Could you also add a test for a limit of 5 (less than 11)?

**Comment by SamSalvatico** [comment]
Yep, I will add it

I set 20 and not 11 just to make it more "realistic", given that usually you set the number of items you need as results for pagination, instead of the number of the items you will receive

**Comment by davidsoderberg** [comment]
Any progress on this PR?

**Comment by Sinister747** [comment]
+1 Appreciate the effort. Would love to see this merged as it's causing me issues in production

**Comment by micheleriva** [comment]
Hi @Sinister747,
@allevo and I will review this again and try to release this by tomorrow. Thanks for the patience

File: packages/docs/open-source/usage/search/facets.md [hunk 2]
```diff
@@ -36,11 +36,11 @@ const results = await search(db, {
   properties: ['description'],
   facets: {
     'categories.primary': {
-      size: 3,
```
Comment (micheleriva) [inline packages/docs/open-source/usage/search/facets.md:39]: @SamSalvatico why changing `size` to `limit`? This is a breaking change

**Review by allevo — APPROVED** [review]
LGTM

**Review by micheleriva — APPROVED** [review]

**Comment by micheleriva** [comment]
@Sinister747 released right now

**Comment by Sinister747** [comment]
> @Sinister747 released right now

Thank you very much, really appreciate it. Enjoy your weekend everyone

**Pull request merged**